src/clds_hash_table.c

```diff
@@ -929,6 +1029,247 @@ CLDS_HASH_TABLE_ITEM* clds_hash_table_find(CLDS_HASH_TABLE_HANDLE clds_hash_tabl
     return result;
 }
 
+CLDS_HASH_TABLE_SNAPSHOT_RESULT clds_hash_table_snapshot(CLDS_HASH_TABLE_HANDLE clds_hash_table, CLDS_HAZARD_POINTERS_THREAD_HANDLE clds_hazard_pointers_thread, CLDS_HASH_TABLE_ITEM*** items, uint64_t* item_count)
+{
+    CLDS_HASH_TABLE_SNAPSHOT_RESULT result;
+
+    if (
+        /* Codes_SRS_CLDS_HASH_TABLE_42_013: [ If clds_hash_table is NULL then clds_hash_table_snapshot shall fail and return CLDS_HASH_TABLE_SNAPSHOT_ERROR. ]*/
+        (clds_hash_table == NULL) ||
+        /* Codes_SRS_CLDS_HASH_TABLE_42_014: [ If clds_hazard_pointers_thread is NULL then clds_hash_table_snapshot shall fail and return CLDS_HASH_TABLE_SNAPSHOT_ERROR. ]*/
+        (clds_hazard_pointers_thread == NULL) ||
+        /* Codes_SRS_CLDS_HASH_TABLE_42_015: [ If items is NULL then clds_hash_table_snapshot shall fail and return CLDS_HASH_TABLE_SNAPSHOT_ERROR. ]*/
+        (items == NULL) ||
+        /* Codes_SRS_CLDS_HASH_TABLE_42_016: [ If item_count is NULL then clds_hash_table_snapshot shall fail and return CLDS_HASH_TABLE_SNAPSHOT_ERROR. ]*/
+        (item_count == NULL)
+        )
+    {
+        LogError("Invalid arguments: CLDS_HASH_TABLE_HANDLE clds_hash_table=%p, CLDS_HAZARD_POINTERS_THREAD_HANDLE clds_hazard_pointers_thread=%p, CLDS_HASH_TABLE_ITEM** items=%p, uint64_t* item_count=%p",
```

Comment:
>** [](start = 167, length = 2)

Seems like this is actually *** after all.
Consider filing an issue to have CLDS_HASH_TABLE_ITEM_HANDLE hiding the CLDS_HASH_TABLE_ITEM*

Reply:
Issue #34 created

---
In reply to: [352379566](https://github.com/dcristoloveanu/clds/pull/33#discussion_r352379566) [](ancestors = 352379566)